packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -421,3 +440,51 @@ export async function userdataMigration(context: MigrationContext): Promise<void
       }
   }
 }
+
+export async function generateApimPluginEnvContent(context: MigrationContext): Promise<void> {
+  // general
+  if (await context.fsPathExists(path.join(".fx", "configs", "projectSettings.json"))) {
+    const projectSettingsContent = fs.readJsonSync(
+      path.join(context.projectPath, ".fx", "configs", "projectSettings.json")
+    );
+    // judge if apim plugin exists
+    let flag_apimPlugin = false;
+    for (const obj of projectSettingsContent["components"])
+      if (obj["name"] === "apim") {
+        flag_apimPlugin = true;
+        break;
+      }
+    // must do this step after configs migration
+    if (flag_apimPlugin) {
+      await context.fsEnsureDir("configs");
```

Comment:
why we need to have a `configs` folder in the project?

Reply:
actually should be pathExists, rather than fsEnsureDir here

Reply:
and as the pathExists judgement is already used at the beginning of this function, so we just need to delete this line.